r/Dockerfile

```diff
@@ -90,4 +75,4 @@ ENV R_LD_LIBRARY_PATH=$LD_LIBRARY_PATH
 
 # build, install, test R package
 CMD ["/bin/bash", "-c", "/arrow/ci/docker_build_cpp.sh && \
-    /arrow/ci/docker_build_r.sh"]
+    /arrow/ci/docker_install_r_deps.sh && /arrow/ci/docker_build_r.sh"]
```

Comment:
The dependencies should be installed in RUN steps to reuse that layer between runs, similarly like:
https://github.com/apache/arrow/blob/4b72c4142a245b3c17eff8acab1095317bf37a13/cpp/Dockerfile#L40-L51